File: algo-py/d-and-c.py
foo = [1, 2, 3, 4]

# find the sum of the provided array using recursion


def bar(arr):

    if arr == []:
        return 0
    else:
        return arr[0] + bar(arr[1:])


print(bar(foo))
print(foo[1:])

File: algo-js/d-and-c.js
let foo = [1, 2, 3, 4];

const sumArray = (arr) => {
  if (arr.length === 0) {
    return 0;
  } else {
    return arr[0] + sumArray(arr.slice(1));
  }
};

console.log(sumArray(foo));
